internal/schemautil/service.go

```diff
@@ -636,19 +636,28 @@ func copyServicePropertiesFromAPIResponseToTerraform(
 		}
 	}
 
-	newUserConfig, err := apiconvert.FromAPI(userconfig.ServiceTypes, serviceType, s.UserConfig)
+	oldUserConfig, err := unmarshalUserConfig(d.Get(serviceType + "_user_config"))
 	if err != nil {
 		return err
 	}
 
-	// Mutates user config in place
-	oldUserConfig := d.Get(serviceType + "_user_config")
-	err = copySensitiveFields(oldUserConfig, newUserConfig)
+	newUserConfig, err := apiconvert.FromAPI(userconfig.ServiceTypes, serviceType, s.UserConfig)
 	if err != nil {
 		return err
 	}
 
-	NormalizeIpFilter(oldUserConfig, newUserConfig)
+	// Apply in-place user config mutations.
+	if len(oldUserConfig)*len(newUserConfig) != 0 {
+		oldUserConfigFirst := oldUserConfig[0]
+
+		newUserConfigFirst := newUserConfig[0]
+
+		// TODO: This should be implemented in the schema returned by /service_types endpoint instead.
```

Comment:
Not sure if we need a TODO for another project here

Reply:
this is a TODO reminder that is supposed to remind us every time when we work with this code that this is not the right way we do things